### ISSUE TITLE
Support string input for number values in color methods.

### ISSUE THREAD

**Issue by RafalFilipek** [styled-components/polished#243]
- `polished` version: 1.7.0

```javascript
transparentize("0.5", "#333") // "rgba(51,51,51,0.5)"
transparentize(0.5, "#333")   // "rgba(51,51,51,0.5)"
lighten("0.5", "#333")        // "#NaNNaNNaN"
lighten(0.5, "#333")          // "#b3b3b3"
```

What you are seeing:

As you can see if `amount` parametr is provided as `string` in `lighten` function you will get `#NaNNaNNaN`.

What you expected to see:

Accept string as a valid parameter type for float parameters or throw error.

Reproduction:

http://jsbin.com/fadevukoji/edit?js,console

Thanks!

**Comment by morajabi** [comment]
@mxstbr Which behaviour do you prefer?

**Comment by mxstbr** [comment]
Couldn't we just `parseFloat(input, 10)` all the inputs and throw an error if it returns NaN or something? I'd rather we err on the side of accepting many inputs, /cc @bhough

**Comment by morajabi** [comment]
@mxstbr Sounds good to me, but one question, why should we parse strings too? Is there any special case?

**Comment by mxstbr** [comment]
Why not? 🤷‍♂️ It's easy to implement, so why not do it? It's nicer for users for sure.

**Comment by morajabi** [comment]
@mxstbr You're right, just thought if it might be bad practice and we don't want to encourage users to do it. But now, I don't think it's bad practice anymore :)

**Comment by bhough** [comment]
We’ve been pretty liberal in allowing most types of inputs and having polished hande them gracefully, so I don’t see any reason not to do the same here.

At some point we should consider a composable way to do this, so we aren’t repeating the same logic in every module for transforming inputs to the proper types.

**Comment by bhough** [comment]
@mxstbr @nikgraf another point of discussion is if we want to continue to provide custom error messaging for incorrect types or start to rely more on Flow/TypeScript to cover that.

**Comment by bhough** [comment]
Addressed in 2.0: https://github.com/styled-components/polished/milestone/1
